### PR TITLE
Realign buttons questionpage

### PR DESCRIPTION
**Pull request by katloboda**
This PR realigns the buttons on the question page depending on the size of the screen. On large screens the generate question button, Qset Name, and all-mine-other button, are all on the same row. On smaller screens, the Qset Name moves above the two buttons.

### CODE REVIEW

**Comment by elliottyates** [comment]
To test this locally:

```
git checkout -b katloboda-realign_buttons_questionpage master
git pull https://github.com/katloboda/askup.git realign_buttons_questionpage
```

**Comment by katloboda** [comment]
Go to this URL to test on Heroku: https://mighty-citadel-8094.herokuapp.com

**Comment by katloboda** [comment]
I fixed Ray's comments about the issues in styling for small screen views (the all-mine-other button would break up into two lines). I fixed this by placing the Generate Question and all-mine-other buttons on two different lines.

File: app/assets/stylesheets/questions.css.scss [hunk 1]
```diff
@@ -1,48 +1,56 @@
-.all-mine-other-filter {
-  margin-right: 25px;  // aligns with right side of question list
-}
-
-// (All-mine-filter relocation at different screen sizes)
-// On small screens (bootstrap xs) show all-mine-other filter
-// just above the question list, since it wraps onto its own .row
-#all-mine-filter-container-xs {
+// On small screens (bootstrap xs) show-small-screen and replace with show-large-screen otherwise
+#show-small-screen {
```
Comment (elliottyates) [inline app/assets/stylesheets/questions.css.scss]: Can we keep this file alphabetical? (roughly) -- e.g. have all-mine-other (a-) entries at the top, move these s-related entries down, etc -- that makes it easier to look for things.

File: app/assets/stylesheets/questions.css.scss [hunk 2]
```diff
@@ -1,48 +1,56 @@
...
-// just above the question list, since it wraps onto its own .row
-#generate-question-button-filter-xs {
-  display: none;
-}
 @media only screen and (max-width: 767px) {
-  #generate-question-button-filter-xs {
-    display: block;
+  
+  #all-mine-other-group {
+    float: none !important;
```
Comment (elliottyates) [inline app/assets/stylesheets/questions.css.scss]: Why is this `float` `!important` stuff needed? maybe a comment here would help, if it is needed, for people like me who aren't real CSS-savvy.

File: app/assets/stylesheets/questions.css.scss [hunk 3]
```diff
@@ -1,48 +1,56 @@
...
+    text-align: center;
   }
-  #generate-question-button-filter {
-    display: none;
+
+  #generate-question-offset {
+    width: 100%;
+
+    a {
+      display: block;
```
Comment (elliottyates) [inline app/assets/stylesheets/questions.css.scss]: Why is this needed? Same thing, comment would be helpful.

File: app/assets/stylesheets/questions.css.scss [hunk 4]
```diff
@@ -1,48 +1,56 @@
...
+#all-mine-other-offset {
+  padding: 20px 25px 10px 0px;
+}
+
+#all-mine-other-group {
+  float: right;
 }
-// End of (generate-question-button-filter relocation at different screen sizes)
+
+#generate-question-offset {
```
Comment (elliottyates) [inline app/assets/stylesheets/questions.css.scss]: I have a few preferences with respect to the ID names. Generally an ID is unique on a page and should be used to identify the element itself (semantically), vs. specifying behavior. Usually when behaviors are the important thing one can use a class tag -- these are not required to be unique (so they're reusable), and by convention a class added to an element clues the developer reading the source that the element has some customized behavior or styling (he/she can then go look in the CSS files to find out how it should behave).

All of the `*-offset` and `show-*` IDs here do look like they're unique, but the IDs aren't semantically useful (I don't know what to expect inside the `#show-small-screen` element -- if it were a `.show-small-screen` class, I would easily be able to ascertain the behavior though). A suggestion would be to split the difference here: 
- make the `show-*` into classes, not IDs (that way you could theoretically use them on other elements to hide/show, and don't need to change the semantic markup of the IDs at all)
- keep the `*-offset`s as IDs, but make the names more semantically helpful, like `#all-mine-other-container` or `#all-mine-other-col` or something. (The idea is that in this case the padding is so specific to that exact element that making it a class wouldn't be very helpful.)

File: app/views/qsets/_show_large_screen.html.erb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<div class = "row">
```
Comment (elliottyates) [inline app/views/qsets/_show_large_screen.html.erb]: please remove spacing around `=`

File: app/views/qsets/_show_small_screen.html.erb [hunk 6]
```diff
@@ -0,0 +1,15 @@
+<div class ="row">
```
Comment (elliottyates) [inline app/views/qsets/_show_small_screen.html.erb]: spacing

File: app/views/qsets/show.html.erb [hunk 7]
```diff
@@ -1,22 +1,15 @@
 <div class="container-fluid" id="qset-show-container" data-qset-id="<%= @qset.id %>">
-  <%= render "qset_context" %>
-  <div id="generate-question-button-filter">
-    <%= render "generate_question" %>
+  <div id ="show-large-screen">
```
Comment (elliottyates) [inline app/views/qsets/show.html.erb]: spacing

File: app/views/qsets/show.html.erb [hunk 8]
```diff
@@ -1,22 +1,15 @@
...
+    <%= render "show_large_screen" %>
   </div>
-  <div id="all-mine-filter-container">
-    <%= render "all_mine_other_filter" %>
+  <div id ="show-small-screen">
```
Comment (elliottyates) [inline app/views/qsets/show.html.erb]: spacing

**Comment by elliottyates** [comment]
Looks good. Thanks for updating mighty-citadel, I was able to test it there. I changed the name of [Math21b](https://mighty-citadel-8094.herokuapp.com/qsets/3) so it has a much longer name, and it looks great at all screen sizes.

File: app/views/qsets/show.html.erb [hunk 9]
```diff
@@ -1,22 +1,15 @@
 <div class="container-fluid" id="qset-show-container" data-qset-id="<%= @qset.id %>">
-  <%= render "qset_context" %>
-  <div id="generate-question-button-filter">
-    <%= render "generate_question" %>
+  <div class="show-large-screen">
+    <%= render "show_large_screen" %>
```
Comment (elliottyates) [inline app/views/qsets/show.html.erb]: the class in the outer div makes sense now, that's great. Why did the partial change to `show_large_screen` though? As a class, `.show-large-screen` clues the developer about the behavior of that element, but as a file/partial name `show_large_screen` isn't very helpful to indicate what's going to be rendered (or what's in the file). I'd switch the filenames back for greater readability.

**Comment by elliottyates** [comment]
This looks great, and thanks for organizing the code so well and for updating mighty-citadel so it's easy to test. Once the filenames are switched back this is ready for a merge!

**Comment by elliottyates** [comment]
:+1:
